locksmith/src/operations/lockOperations.ts

```diff
@@ -74,3 +75,139 @@ export async function getKeyHolderMetadata(
     },
   })
 }
+
+export async function getIcon(address: string) {
+  /**
+   * This computes how much rotation to apply to the lock glyph
+   * @param {string} address
+   * @returns {number}
+   */
+  const degreesOfRotation = (address: string) => {
+    const n = parseInt(address)
+    return (n % 36) * 10
+  }
+
+  const translateAndScale = (address: string) => {
+    const n = parseInt(address)
+    return n % 2 == 0 ? '' : 'tranlate(216, 0) scale(-1, 1)'
+  }
+
+  // Default colors
+  let colors = ['#8c8c8c', '#e8e8e8', '#c3c3c3']
+  const scheme = new ColorScheme()
+  const mainColor = address.substring(2, 8).toUpperCase()
+  scheme
+    .from_hex(mainColor)
+    .scheme('triade')
+    .variation('pastel')
+  colors = scheme.colors().map((c: string) => `#${c}`)
+
+  const originalIcon = [
```

Comment:
There are quite a bit of constants (eg: stripedIcon,chomIcon,bitIcon) it may be useful to provide some comments on why they are meaningful   being leveraged in this code, it